### PR TITLE
Fix several LSP completion kind mappings

### PR DESCRIPTION
**Pull request by DoctorKrolic**
Fixes: https://github.com/dotnet/roslyn/issues/60755

Also fixed mapping for namespaces

### CODE REVIEW

File: src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs [hunk 1]
```diff
@@ -51,7 +51,7 @@ internal static class ProtocolConversions
             { WellKnownTags.Assembly, LSP.CompletionItemKind.File },
             { WellKnownTags.Class, LSP.CompletionItemKind.Class },
             { WellKnownTags.Constant, LSP.CompletionItemKind.Constant },
-            { WellKnownTags.Delegate, LSP.CompletionItemKind.Method },
+            { WellKnownTags.Delegate, LSP.CompletionItemKind.Delegate },
             { WellKnownTags.Enum, LSP.CompletionItemKind.Enum },
             { WellKnownTags.EnumMember, LSP.CompletionItemKind.EnumMember },
             { WellKnownTags.Event, LSP.CompletionItemKind.Event },
```
Comment (dibarbet) [inline src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs:57]: I think we can also add the other new mapping for `CompletionItemKind.ExtensionMethod` on the line below this one.

I'll leave it up to you if you want to add in this PR, or I can do it later.

**Review by dibarbet — APPROVED** [review]
much thanks!

File: src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs (hunk 1 above)
Comment (DoctorKrolic) [reply src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs:57]: Yeah, I messed that one

**Comment by DoctorKrolic** [comment]
@dibarbet Merge this PR please

**Comment by dibarbet** [comment]
done, thanks!

**Pull request merged**